python_modules/dagster/dagster/_daemon/daemon.py

```diff
@@ -260,10 +261,37 @@ def core_loop(
 
 
 class SensorDaemon(DagsterDaemon):
+    def __init__(self, settings: Mapping[str, Any]) -> None:
+        super().__init__()
+        self._exit_stack = ExitStack()
```

Comment:
nit: typing + remove `else` clause

```suggestion
        self._exit_stack = ExitStack()
        self._threadpool_executor: Optional[ThreadPoolExecutor] = None
        self._submit_threadpool_executor: Optional[ThreadPoolExecutor] = None
```